Chore: consolidate CCL release 3.3.x

mlenkeit opened the pull request:
This PR consolidates the changes for CCL 3.3.x in a separate branch, e.g. to leverage the OS-specific integration tests and to consolidate the UA review.

T-Bendrath committed:
feat(ccl): add API draft of getDccAdmissionCheckScenarios

T-Bendrath committed:
test(ccl) corrected assertions tests for admission scenarios

T-Bendrath committed:
test(ccl) fix iteration in scenario.subtitleText check

T-Bendrath committed:
feat(ccl) changed admission scenario return types of identifier, enabled

T-Bendrath committed:
fix(ccl) Fixed ID of admission-check-scenarios schema

T-Bendrath committed:
feat(ccl) Added 'DE' in admission-check-scenarios API

T-Bendrath committed:
refactor(ccl) Ease configuration of scenarios in admission-check-scenarios API

T-Bendrath committed:
feat(ccl) Hide subtitleText property if disabled in admission-check-scenarios API

T-Bendrath committed:
test(ccl) Added test to validate empty subtitleTexts in admission-check-scenarios API

T-Bendrath committed:
Merge branch 'cwa/release/2.19' into feat/get-dcc-admission-check-scenarios-api

T-Bendrath committed:
chore(ccl): rename i18n field descriptor for admission scenario

T-Bendrath committed:
chore(ccl): changed function sequences to comply with lexicographical sorting

T-Bendrath committed:
refactor(ccl): avoid unnecessary complexity in API implementation

T-Bendrath committed:
Merge pull request #36 from corona-warn-app/feat/get-dcc-admission-check-scenarios-api

Feat: Create CCL API getDCCAdmissionCheckScenarios (EXPOSUREAPP-11990)

mlenkeit committed:
Merge branch 'cwa/release/2.19' of https://github.com/corona-warn-app/cwa-app-ccl into cwa/release/2.19

mlenkeit committed:
Merge branch 'main' into cwa/release/2.19

mlenkeit committed:
chore: export test cases for admission check scenarios

mlenkeit committed:
chore(schema): add scenarioIdentifier

mlenkeit committed:
test(ccl): scenarioIdentifier is reflected in admissionState

mlenkeit committed:
feat(ccl): reflect scenarioIdentifier in admissionState

mlenkeit committed:
test(ccl): check scenario-specific badge texts

T-Bendrath committed:
feat(ccl): added basic check for a non-existent scenario

T-Bendrath committed:
feat(ccl): added scenario specific badgeText in walletInfo

mlenkeit committed:
Merge pull request #41 from corona-warn-app/feat/support-scenario-in-wallet-info

Feat: dummy-support for scenarioIdentifier (testing-only) (EXPOSUREAPP-11991)

mlenkeit committed:
chore(schema): allow full semver in Version

mlenkeit committed:
chore: add pre-push hook

mlenkeit committed:
feat(ccl): add maskState to schema

mlenkeit committed:
test(ccl): add assertions for maskState

mlenkeit committed:
chore(i18n): add dummy texts for maskState

mlenkeit committed:
feat(ccl): add dummy logic for maskState

mlenkeit committed:
feat(ccl): add maskState OTHER

mlenkeit committed:
fix(ccl): skip texts for maskState OTHER

mlenkeit committed:
Merge branch 'release/3.2.x' into testing-only/mask-states

mlenkeit committed:
Revert "chore: add pre-push hook"

This reverts commit 5911c73c41154f6757a11b35f23a174da3312cfe.

mlenkeit committed:
Merge branch 'release/3.3.x' into testing-only/mask-states

mlenkeit committed:
Merge branch 'cwa/release/2.19' into testing-only/mask-state-by-state

mlenkeit committed:
test(ccl): mask is always required in Hamburg

mlenkeit committed:
docs: annotate with JS

thomasaugsten committed:
first draft mask rules

thomasaugsten committed:
add other state

thomasaugsten committed:
remove partial dob

thomasaugsten committed:
Merge branch 'release/3.2.x' into testing-only/mask-state-by-state

# Conflicts:
#	lib/ccl/functions/__evaluateBoosterNotificationRules.js

thomasaugsten committed:
add other with admission state

thomasaugsten committed:
visible false admission state

thomasaugsten committed:
add new scenario ids

thomasaugsten committed:
add scenario list

thomasaugsten committed:
chore: fix typo

mlenkeit committed:
Merge branch 'testing-only/mask-state-by-state' into feature/display-mask-state

mlenkeit committed:
test(ccl): add test cases for mask state

mlenkeit committed:
chore: remove only-statements from tests

mlenkeit committed:
Revert "visible false admission state"

This reverts commit 60197ec5f8b916d9c81e578deec871dc4ffb9846.

mlenkeit committed:
chore: remove trailing whitespace

mlenkeit committed:
test(ccl): expect all states as scenarios

mlenkeit committed:
feat(ccl): display each state as scenario

mlenkeit committed:
feat(ccl): determine mask state

mlenkeit committed:
Revert "docs: annotate with JS"

This reverts commit 8fdfd8d6dca5646ada0c3a9a2f65d186a850ab00.

mlenkeit committed:
chore: remove manual EN translations

mlenkeit committed:
test(ccl): always except admissionState OTHER

mlenkeit committed:
feat(ccl): drop logic for different admission states

mlenkeit committed:
chore(i18n): update text for admission state OTHER

mlenkeit committed:
test(cc): expect TC validity of 24 hours

mlenkeit committed:
feat(ccl): set TC validity to 24 hours

mlenkeit committed:
test(ccl): expect TCs as first verification certificate

mlenkeit committed:
feat(cc): use TCs as first verification certificate

mlenkeit committed:
Merge pull request #102 from corona-warn-app/feature/display-mask-state

Feature: Determine mask state (EXPOSUREAPP-13911)

mlenkeit committed:
Merge pull request #103 from corona-warn-app/feature/hide-admission-state

Feature: Remove admission state (EXPOSUREAPP-13914, EXPOSUREAPP-13916)

mlenkeit commented on resources/i18n/de/strings.xml:
```diff
@@ -35,9 +81,9 @@
   <!-- XTXT: Admission State - OTHER Card Body -->
-  <string name="ADMISSION_STATE_OTHER_LONG_TEXT">"Ihre Zertifikate erfüllen derzeit keine Regel."</string>
+  <string name="ADMISSION_STATE_OTHER_LONG_TEXT">"Der G-Status ist zur Zeit nicht relevant und wird daher nicht von der App ausgewiesen."</string>
   <!-- XTXT: Admission State - OTHER State Change Notification Text -->
-  <string name="ADMISSION_STATE_OTHER_SCN_TEXT">"Ihre Zertifikate erfüllen derzeit keine Regel."</string>
+  <string name="ADMISSION_STATE_OTHER_SCN_TEXT">"Der G-Status ist zur Zeit nicht relevant und wird daher nicht von der App ausgewiesen."</string>
```
These texts need double-checking, AFAIK we haven't used _G-Status_ so far in the app.

KathrinLuecke replied:
How about: "Zertifikate werden zur Zeit nicht geprüft und Ihr Status daher nicht von der App ausgewiesen."

mlenkeit committed:
test(ccl): expect scenario in mask state subtitle

mlenkeit committed:
feat(ccl): display scenario in mask state subtitle

mlenkeit committed:
test(ccl): update test cases for vaccination state based on IfSG

there are now the following states:

- COMPLETE_IMMUNIZATION for anything that is full immunization based on
IfSG (e.g. 3 shots) and where the last event is not a recovery
- COMPLETE_IMMUNIZATION_RC for anything that is full immunization based
on IfSG (e.g. 2 shots + recovery) and where the last event is not a
recovery
- COMPLETE_IMMUNIZATION_RC_PENDING for anything that matches
COMPLETE_IMMUNIZATION_RC within the first 29 days
- COMPLETE_BOOSTER for anything that is beyond COMPLETE_IMMUNIZATION or
COMPLETE_IMMUNIZATION_RC
- PARTIAL_IMMUNIZATION for anything else

mlenkeit committed:
test(ccl): expect only federal state name in maskState subtitle

mlenkeit committed:
test(ccl): expect scenario-dependent long text for mask state

mlenkeit committed:
test(ccl): assert complete mask state subtitle

mlenkeit committed:
test(ccl): update expected admission check scenario texts

mlenkeit committed:
test(ccl): update expected admission state long text

mlenkeit committed:
chore(i18n): update texts for admission scenario selection

mlenkeit committed:
chore(i18n): update long text for admission state

mlenkeit committed:
feat(ccl): display only federal state name in mask state subtitle

mlenkeit committed:
feat(ccl): display scenario-dependent long text for mask state

mlenkeit committed:
test(ccl): remove trailing whitespace from expected texts

mlenkeit committed:
chore(i18n): drop redundant mask state texts

mlenkeit committed:
refactor(ccl): reuse mask state title text for badge

mlenkeit committed:
feat(ccl): align vaccination state with IfSG

mlenkeit committed:
chore(i18n): drop redundant vaccination state texts

mlenkeit committed:
fix(ccl): remove repeating text chunks

mlenkeit committed:
Merge pull request #108 from corona-warn-app/feat/update-admission-check-scenario-texts

Chore: Update texts for admission check scenario selection (EXPOSUREAPP-13911)

mlenkeit committed:
Merge pull request #109 from corona-warn-app/feature/update-admission-state-text

Chore: Update texts for admission state (EXPOSUREAPP-13911)

mlenkeit committed:
Merge pull request #106 from corona-warn-app/feat/show-selected-state-in-mask-state

Feature: Display selected federal state in texts for mask state (EXPOSUREAPP-13911)

mlenkeit committed:
Merge pull request #107 from corona-warn-app/feature/update-vaccination-state

Feature: Determine vaccination state based on latest IfSG (EXPOSUREAPP-13915)

mlenkeit committed:
chore(i18n): drop redundant admission state texts

mlenkeit committed:
refactor(ccl): reuse admission state long text for notification

mlenkeit committed:
refactor(ccl): simplify lookup of admission state texts

mlenkeit committed:
refactor(ccl): use constant for federal state ids

Ein-Tim commented on lib/ccl/ccl-constants.js:
```diff
@@ -1,5 +1,7 @@
-export const TC_PCR_VALIDITY_IN_HOURS = 72
-export const TC_RAT_VALIDITY_IN_HOURS = 48
+export const TC_PCR_VALIDITY_IN_HOURS = 24
+export const TC_RAT_VALIDITY_IN_HOURS = 24
```
Shouldn't this change be back ported to the apps "Receive test results" feature (e.g. show RAT as out of date after 24h)?

mlenkeit replied:
This is not yet decided.

Ein-Tim replied:
@mlenkeit 

Ok, thanks, but it is in the talks?

mlenkeit replied:
Yes, but it will most likely stay as it is at the moment.

Ein-Tim commented:
I have a question reg. https://github.com/corona-warn-app/cwa-app-ccl/pull/107, if the texts & the immunization state in the app changes to partial immunization only, will there be a booster recommendation notification which not only **recommends** the booster but clearly states that a booster is **necessary** now?

mlenkeit committed:
chore(actions): run test cases in chunks

this helps to prevent out of memory errors in gradle due to large JSON files of test cases that are passed at once

mlenkeit committed:
Merge branch 'main' into chore/consolidate-release-3.3.x

mlenkeit committed:
refactor(ccl): remove dead code

mlenkeit committed:
refactor(ccl): simplify lookup of vaccination state texts

mlenkeit committed:
refactor(ccl): drop redundant test certificate analysis

mlenkeit committed:
chore(actions): use JFN feature branch instead of main

mlenkeit committed:
chore: use common functions for CCL test case export

mlenkeit committed:
chore: use common functions in dist tests

mlenkeit committed:
refactor(ccl): simplify vaccination state logic

mlenkeit committed:
refactor(ccl): simplify mask state logic

mlenkeit committed:
refactor(ccl): drop redundant admission state artefacts

mlenkeit committed:
refactor(ccl): drop assertions for hasBoosterEquivalent

mlenkeit committed:
refactor(ccl): drop assertions for hasBooster

mlenkeit committed:
refactor(ccl): drop logic for hasBoosterEquivalent

mlenkeit committed:
refactor(ccl): drop logic for hasBooster

mlenkeit committed:
refactor(ccl): drop __hasPreviousFullImmunization

mlenkeit committed:
Merge pull request #110 from corona-warn-app/refactor/drop-redundant-logic

Refactor: drop redundant logic and text

mlenkeit committed:
Revert "chore(actions): run test cases in chunks"

This reverts commit 5d4f4e4e799fcb38e1ff7a573a7d05efc0d3534e.

mlenkeit committed:
chore(i18n): add 3G+ to list of status

KathrinLuecke reviewed (approved):
UA approved

KathrinLuecke reviewed (approved):
UA approved

mlenkeit committed:
Revert "chore(actions): use JFN feature branch instead of main"

This reverts commit 1f4432cac90410b654c6dd540d06058b0ed28a5c.

mlenkeit committed:
Merge pull request #111 from corona-warn-app/chore/reduce-test-case-footprint

Chore: Reduce test case footprint

mlenkeit commented:
> I have a question reg. #107, if the texts & the immunization state in the app changes to partial immunization only, will there be a booster recommendation notification which not only **recommends** the booster but clearly states that a booster is **necessary** now?

@Ein-Tim this is not planned at the moment.

Ein-Tim commented:
@mlenkeit 

So the user will just loose their fully vaccinated status without further information in the app why?

I think more information should be given.

MikeMcC399 commented:
@Ein-Tim 
 
> So the user will just loose their fully vaccinated status without further information in the app why?
> 
> I think more information should be given.

This is more the responsibility of the new features text in the app e.g. for Android 2.27.x https://github.com/corona-warn-app/cwa-app-android/blob/release/2.27.x/Corona-Warn-App/src/main/res/values/release_info_strings.xml.

At the moment only the mask rule change is mentioned in the release info. It would be good for the release info to also explain that the 2G, 2G+, 3G and 3G+ status is removed and that the rules in general have been updated to correspond to current regulations.

Also the blog text for the release 2.27 would have the opportunity to explain about this.

This is all work-in-progress, so I would wait and see what develops.

Ein-Tim commented:
@MikeMcC399 

I'll take your advice and look at it again in a week or so.

Ein-Tim commented:
Okay, question answered with https://github.com/corona-warn-app/dcc-rule-translation/issues/10#issuecomment-1252418923.
